### ISSUE TITLE
Improve documentation

### ISSUE THREAD

**Issue by dbader** [dbader/schedule#4]
Currently the module only comes with a README and docstrings within the code. We should make documentation more explicit by providing the following:
- Full HTML documentation generated from the source, hosted on read-the-docs.org
  - [read-the-docs tutorial](https://read-the-docs.readthedocs.org/en/latest/getting_started.html)
- An improved rationale why this module exists
- Example code for common use cases
- A FAQ section that deals with a few points raised in the [reddit /r/python thread](http://www.reddit.com/r/Python/comments/1f6s7s/schedule_python_job_scheduling_for_humans/)

**Comment by karan** [comment]
Would really appreciate example code for multiple use cases:

- Clear logs every day
- Run task every minute
- Run task every minute without blocking anything
- Run task from a message queue (redis)
- Run in a distributed env
- Handling exceptions the right way

**Comment by DieterKoblenz** [comment]
Please do, love this module but documentation so far is very poor.

**Comment by dbader** [comment]
I think this is still a key thing we should focus on to improve schedule. Bump.

**Comment by dbader** [comment]
Alright I just merged #108 – docs are live now at https://schedule.readthedocs.io

I'd greatly appreciate some help with fleshing out the docs further. I've laid down some basic structure but haven't migrated the FAQ or continued building it out.

If you're not comfortable with editing the Sphinx docs then improving the docstrings in the code directly would help a lot.

Better docs is one of the best things we can do for the project right now.

**Comment by dbader** [comment]
I want to link some docs-related stuff for future reference:
* http://www.sphinx-doc.org/en/1.4.8/tutorial.html
* https://daler.github.io/sphinxdoc-test/includeme.html
* http://www.sphinx-doc.org/en/1.4.8/ext/autodoc.html

I also haven't updated the dev docs on generating and viewing the Sphinx web docs locally. Basically you need to:

```
pip install -r requirements-dev.txt
cd docs
make html
cd _build/html
python -m SimpleHTTPServer 8888
```

and then open http://localhost:8888 in your favorite browser 🌟
